src/MarlinSimulator/application.cpp

```diff
@@ -72,67 +75,157 @@ Application::Application() {
+        ImPlot::SetNextPlotLimitsY(0.0f, 1.2f, ImGuiCond_Always);
+        static int rt_axis = ImPlotAxisFlags_NoTickLabels | ImPlotAxisFlags_LockMin;
+        if (ImPlot::BeginPlot("##Scrolling", "Time (ns)", NULL, ImVec2(-1,150), ImPlotAxisFlags_NoTickLabels | ImPlotFlags_Query, rt_axis, rt_axis)) {
+          ImPlot::PlotLine("pin", &sdata.Data[0].x, &sdata.Data[0].y, sdata.Data.size(), sdata.Offset, sizeof(ImPlotPoint));
+          ImPlot::EndPlot();
         }
-        if (p.pin == monitor_pin) ImGui::SetItemDefaultFocus();
       }
-      ImGui::EndCombo();
-    }
 
-    if (Gpio::pin_map[monitor_pin].event_log.size()) {
-      ScrollingData sdata;
+      static bool export_single_pin = false;
+      if (ImGui::Button("Export selected pin to file")) {
+        export_single_pin = true;
+        ImGuiFileDialog::Instance()->OpenDialog("PulseExportDlgKey", "Choose File", "Value Change Dump (*.vcd){.vcd},.*", ".");
+      }
+
 
-      pin_log_data last{};
-      for (auto v : Gpio::pin_map[monitor_pin].event_log) {
-        if (last.timestamp) sdata.AddPoint(v.timestamp, last.value);
-        sdata.AddPoint(v.timestamp, v.value);
-        last = v;
+      if (ImGui::Button("Export pins matching regex to file")) {
+        export_single_pin = false;
+        ImGuiFileDialog::Instance()->OpenDialog("PulseExportDlgKey", "Choose File", "Value Change Dump (*.vcd){.vcd},.*", ".");
+      }
+
+      static char export_regex[128] = "";
+      ImGui::SameLine();
+      ImGui::InputText("Pin regex", export_regex, sizeof(export_regex));
+
+      if (ImGuiFileDialog::Instance()->Display("PulseExportDlgKey", ImGuiWindowFlags_NoDocking))  {
+        try{
+        if (ImGuiFileDialog::Instance()->IsOk()) {
+          std::string image_filename = ImGuiFileDialog::Instance()->GetFilePathName();
+
+          using namespace vcd;
+
+          HeadPtr head = makeVCDHeader(static_cast<TimeScale>(50), TimeScaleUnit::ns, utils::now());
+          VCDWriter writer{image_filename, std::move(head)};
+
+          if (export_single_pin) {
+            std::string pin_name(active_label);
+            auto scope = pin_name.substr(0, pin_name.find_first_of('_'));
+            auto var = writer.register_var(scope, pin_name, VariableType::wire, 1);
+
+            if (Gpio::pin_map[monitor_pin].event_log.size() && pin_array[monitor_pin].is_digital) {
+              for (const auto &value : Gpio::pin_map[monitor_pin].event_log) {
+                writer.change(var, value.timestamp / 50, utils::format("%u", value.value));
+              }
+            }
+          }
+          else {
+            std::map<size_t, VarPtr> pin_to_var_map;
+            std::regex expression(export_regex);
+
+            for (auto pin : pin_array) {
+              std::string pin_name(pin.name);
+              bool regex_match = strlen(export_regex) == 0 || std::regex_search(pin_name, expression);
+              auto scope = pin_name.substr(0, pin_name.find_first_of('_'));
```

Comment:
I attempted to scope the lines based on the first portion of the pin name prior to `_`. This doesn't seem to actually work when imported into PulseView. I'm not sure if that is me doing something wrong or a PulseView limitation.

The current code doesn't seem to be hurting anything, but we could probably also eliminate it and use a constant string with no discernable difference.

(I'll wait to hear an opinion before changing anything here.

Reply:
This is what this ends up doing right now. I've exported pins matching `_(STEP|DIR)_` to demonstrate the grouping I attempted:

```
$scope module E0 $end
$var wire 1 0 E0_DIR_PIN $end
$var wire 1 1 E0_STEP_PIN $end
$upscope $end
$scope module E1 $end
$var wire 1 2 E1_DIR_PIN $end
$var wire 1 3 E1_STEP_PIN $end
$upscope $end
$scope module X $end
$var wire 1 4 X_DIR_PIN $end
$var wire 1 5 X_STEP_PIN $end
$upscope $end
$scope module Y $end
$var wire 1 6 Y_DIR_PIN $end
$var wire 1 7 Y_STEP_PIN $end
$upscope $end
$scope module Z $end
$var wire 1 8 Z_DIR_PIN $end
$var wire 1 9 Z_STEP_PIN $end
$upscope $end
```

Reply:
I had hoped it would group the signals like this in PulseView, but it does not (I manually re-grouped these):

![image](https://user-images.githubusercontent.com/20053467/144725717-e995260b-3e1e-41c7-8a2c-bd67d72c551b.png)